tedpermission/src/main/java/com/gun0912/tedpermission/TedPermissionActivity.java

```diff
@@ -158,7 +159,7 @@ public void onClick(DialogInterface dialogInterface, int i) {
 
     private void checkPermissions(boolean fromOnActivityResult) {
 
-        ArrayList<String> needPermissions = new ArrayList<>();
+        List<String> needPermissions = new ArrayList<>();
```

Comment:
Is there a reason to change this?
It is a local variable and there is no connection point with the outside.

Reply:
It was just base on `program to interfaces, not implementations`.

Although this request changed the listener used by the` ArrayList` to `List`, it was consistent under the same principle.

Reply:
@dotkebi

I think #75 means that allow all `List`, not limited to `ArrayList`.
Because of, user can use any `List` type. (`ArrayList` or `LinkedList`...)
In this case, however,
there is no reason to allow another `List`.
It is certain that it is `ArrayList`, and there is no connection point with the outside.

If you have any other comments, please share them!!
Thanks!

Reply:
@JSpiner 

* Why `List <String>`
Only use `add` method of` Collection`. There is no part that uses method implemented in ArrayList in particular.

Therefore, it can only use `Collection` in your method implementation, but the requestPermissions signature has to be cast to` List <String> `, so if I limits it to a minimum condition, `List` is most appropriate.

* Type casting problems,
First, checkPermissions (boolean fromOnActivityResult) is hard to override because it is `private`.
It is a type to pass to the parameter of `requestPermissions`, so I think it is not necessary to consider it.

Reply:
@dotkebi 
Now I agree with you.
Thank you for the kind explanation!
I'll change it to approved.

Thanks!